src/Aspire.Hosting.Qdrant/QdrantServerResource.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a Qdrant database.
+/// </summary>
+public class QdrantServerResource : ContainerResource, IResourceWithConnectionString
+{
+    internal const string PrimaryEndpointName = "http";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QdrantServerResource"/> class.
+    /// </summary>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="apiKey">A <see cref="ParameterResource"/> that contains the API Key</param>
+    public QdrantServerResource(string name, ParameterResource apiKey) : base(name)
+    {
+        ArgumentNullException.ThrowIfNull(apiKey);
+        ApiKeyParameter = apiKey;
+    }
+
+    private EndpointReference? _primaryEndpoint;
+
+    /// <summary>
+    /// Gets the parameter that contains the Qdrant API key.
+    /// </summary>
+    public ParameterResource ApiKeyParameter { get; }
+
+    /// <summary>
+    /// Gets the primary endpoint for the Qdrant database.
+    /// </summary>
+    public EndpointReference PrimaryEndpoint => _primaryEndpoint ??= new(this, PrimaryEndpointName);
+
+    /// <summary>
+    /// Gets the connection string expression for the Qdrant database.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression =>
+       ReferenceExpression.Create(
+            $"http://{PrimaryEndpoint.Property(EndpointProperty.Host)}:{PrimaryEndpoint.Property(EndpointProperty.Port)}");
```

Comment:
> http://

This seems wrong. I don't think we should be hard-coding the scheme in the Expression.

Reply:
that's right, the scheme is part of the endpoint reference:

PrimaryEndpoint.Property(EndpointProperty.Scheme)

Reply:
@eerhardt How do we handle resources that don't have a TLS certificate by default? That's the situation here. https://qdrant.tech/documentation/guides/security/#tls

Reply:
We can't do https here but it should use the scheme defined by the endpoint.

Reply:
changed to scheme already in latest PR feedback.  As an aside, SSL could be enabled using dotnet dev-certs but felt hacky (but got it working).  Anyhow, PR update to use scheme